src/components/HistoricalForecastWidget/HistoricalForecastWidget.tsx

```diff
@@ -0,0 +1,109 @@
+import { useEffect, useState } from 'react';
+import { format, fromUnixTime } from 'date-fns';
+import pick from 'lodash/pick';
+import CircularProgress from '@mui/material/CircularProgress';
+import Tooltip from '@mui/material/Tooltip';
+import { useSnackbar } from 'notistack';
+import { FormValuesType } from 'components/HistoricalForecastForm/validation';
+import ForecastBody from 'types/forecastBody';
+import findCountryNameByCode from 'utils/findCountryNameByCode';
+import formatTemperatureData from 'utils/formatTemperature';
+import FORECAST from './forecast';
+import WEEK_DAYS from './weekDays';
+
+interface Props {
+  formValues: FormValuesType;
+}
+
+const SPINNER_SIZE = 25;
+
+export default function HistoricalForecastWidget({ formValues }: Props) {
+  const { enqueueSnackbar } = useSnackbar();
+  const [forecast, setForecast] = useState<ForecastBody[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    console.log(formValues);
+    const fetchForecast = () => {
+      try {
+        setIsLoading(true);
+        const { list } = FORECAST;
+        const forecastData = list.map((data) =>
+          pick(data, ['dt', 'main', 'weather']),
+        );
+        console.log(forecastData);
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        setForecast(forecastData);
```

Comment:
As you recall you need to create two-dimensional array. Inner arrays will hav 7 objects - weather data for each day of weekday
In order to achieve that you need to reduce an array. If a weekday is Monday, you need to add a new array to accumulator and push all objects to created array unless predicate (i.e. function that performs `.reduce`) reaches new Monday.
Also, don't forget to fill the first week by placeholder values if weekday of the first element in the first row starts not from Monday